qutebrowser/completion/completionwidget.py

```diff
@@ -255,3 +261,25 @@ def showEvent(self, e):
         if scrollbar is not None:
             scrollbar.setValue(scrollbar.minimum())
         super().showEvent(e)
+
+    def _open_completion_if_needed(self):
+        if not config.get('completion', 'auto-open'):
+            self._set_auto_open(True)
+
+    def _set_auto_open(self, enabled):
+        cmd = objreg.get('status-command', scope='window', window=self._win_id)
+
+        if enabled:
```

Comment:
If you press `<Tab>` multiple times (with `auto-open = False`), this will cause the signals to be connected again every time, thus calling `cmd.update_completion` multiple times every time there's a change.

Shouldn't the `connected` conditional below be here instead, so all this is only ran if it wasn't run before?